Bugfix/accuracy tests

leoTiez opened the pull request.

leoTiez committed:
Update: Changed parameters and added greyscale flag

leoTiez committed:
Update: Added greyscale flag for cifar data sets

leoTiez committed:
Update: Investigation of the recurrent layer

Layer is slightly worse than without any recurrent input.
Probably problem with bias term.

leoTiez committed:
Bugfix: Removed return of subset

leoTiez committed:
Update: Removed unnecessary loop for gpu

The loop could be removed since a flag solved the issue of dead
ends

leoTiez committed:
Update: Changed different spelling of gray to be consistent

leoTiez committed:
Update: Tests and tweaking of parameters

leoTiez committed:
'Update: Added parameters to determine while loop behavior

leoTiez committed:
Update: Changed the depeth for cifar10 back to 3

leoTiez committed:
Update: Changed the optimiser to gradient descent

leoTiez committed:
Update: parameter tweaking, increasing epochs and learning rate

leoTiez committed:
Bugfix: Compute cell states after first output computation

The cell states are calculated with the feedforward
output first

leoTiez committed:
Update: Change of hyperparameters for better learning performance

leoTiez committed:
Update: Clean up

Removed unnecessary parameters and comments

leoTiez committed:
Update: Implemented adaptive learnign rate

Learning rate is  now dynamically passed
When the accuracy does not increase for a number of epochs,
it is decreased by multiplying by a constant factor.
If adpative learning factor is set 1, no adaptive learning
is performed.

Updated docstrings

bt-s commented on RCNN_tf.py:
```diff
@@ -23,10 +22,11 @@
 PRECISION_TF = tf.float32
 PRECISION_NP = np.float32
 PADDING_LIST = ['SAME', 'VALID']
+initializer = tf.contrib.layers.variance_scaling_initializer()
```
Good. Maybe we could capitalize it macro style, but this is also fine!

leoTiez replied:
No, that's true. Thanks for pointing it out!

bt-s commented on RCNN_tf.py:
```diff
@@ -104,97 +81,80 @@ def rcl(input_data, num_input_chans, num_filter, filter_shape, num_of_data,
+        return x, recurrent_states, output
 
-        return output
+    # TODO I am not sure why but this learns quicker. Investigate?
```
It probably has to do with the order in which Tensorflow calls the functions in the while loop.
We could investigate it, but if you have seen that it is faster I think we could just leave it as it is.

leoTiez replied:
Yea think so, too. For the moment we can leave it as it is

bt-s commented on RCNN_tf.py:
```diff
@@ -104,97 +81,80 @@ def rcl(input_data, num_input_chans, num_filter, filter_shape, num_of_data,
         padding='SAME'
     )
 
-    if "CPU" in processing_unit.upper():
```
Nice catch that we do not have to specify anymore whether we are using a GPU or a CPU!

bt-s commented on RCNN_tf.py:
```diff
@@ -203,13 +163,12 @@ def convolutional_layer(input_data, num_input_chans, num_filter, filter_shape,
+                         num_filter]
     strides = [1, stride[0], stride[1], 1]
 
     initializer = tf.contrib.layers.variance_scaling_initializer()
```
Do we need to declare the initializer again over here? Can we not just declare it globally?

leoTiez replied:
Yea, you are right, I totally forgot that line. I will remove that!

bt-s commented on load_data.py:
```diff
@@ -65,17 +79,22 @@ def data_loader(dataset, file_name, dtype):
 
-        data = preprocess_cifar_imgs_(np.asarray(data_dict['data'], dtype=dtype))
+        data = preprocess_cifar_imgs_(np.asarray(data_dict['data'], dtype=dtype),
+                                      dtype=dtype,
```
Is this ```dtype=dtype``` necessary?

leoTiez replied:
Yes, it is necessary for converting to grayscale

bt-s commented on load_data.py:
```diff
@@ -65,17 +79,22 @@ def data_loader(dataset, file_name, dtype):
 
-        data = preprocess_cifar_imgs_(np.asarray(data_dict['data'], dtype=dtype))
+        data = preprocess_cifar_imgs_(np.asarray(data_dict['data'], dtype=dtype),
+                                      dtype=dtype,
```
What about this Is this ```dtype=dtype```; is that also necessary?

leoTiez replied:
Same, necessary for the conversion to grayscale

bt-s reviewed (approved):
This looks very solid: I like the implementation!

adrianc-a reviewed (approved):
lgtm

leoTiez committed:
Update: Changed default parameter for adaptive learning factor

leoTiez committed:
Update: Changed hyperparameter of mnist according to changes in cifar

leoTiez committed:
Update: Passing dropout rate as a parameter

Instead of setting it statically to 0.2, the dropout
rate can be passed as a parameter to the function train

leoTiez committed:
Merge pull request #29 from leoTiez/feature/adaptiveLearningRate

Feature/adaptive learning rate

leoTiez committed:
Update: Replaced re initialization of global function

The function for the initialization has been set to a global
variable. Thus, re initialization is not necessary.
The concerning lines have been removed.

Spelling has been changed to capital letters

leoTiez commented:
solves #13 and #24
